src/Compilers/CSharp/Test/Semantic/FlowAnalysis/RegionAnalysisTests.cs

```diff
@@ -3129,6 +3129,103 @@ static void Main(string[] args)
             Assert.Equal("args, i", GetSymbolNamesJoined(analysis.WrittenOutside));
         }
 
+        [Fact]
+        public void TestDataFlowsInCtorInit()
+        {
+            var analysis = CompileAndAnalyzeDataFlowConstructorInitializer(@"
```

Comment:
>CompileAndAnalyzeDataFlowConstructorInitializer [](http://example.com/codeflow?start=27&length=47)

We need to make sure to cover all public entry points that are supposed to change behavior and all added public entry points. It looks like this helper doesn't call any of them.

Reply:
It calls the SemanticModel.AnalyzeDataFlow(SyntaxNode) - which does not throw an exeption anymore for ConstructorInitializerSyntax.

Reply:
I tried to implement as close as possible to the other tests - but you are right the non-dispatching csharp extension method is not called. Should we execute the analyses twice for the extension method and for the general method?

Reply:
> It calls the SemanticModel.AnalyzeDataFlow(SyntaxNode) - which does not throw an exeption anymore for ConstructorInitializerSyntax.

I think that API is internal. 

> you are right the non-dispatching csharp extension method is not called. Should we execute the analysis twice: for the extension method and for the general method?

I do not think this is necessary. It should be sufficient to a have a single test that explicitly covers all relevant public entry points.

Reply:
Where should those tests located?

Reply:
Are there some similar tests for Statements / Expressions? I would add the next to those.

Reply:
> Where should those tests located?

This file is fine.